Fix lesson2

dydy-jora opened the pull request.

dydy-jora committed:
Add files via upload

ДЗ №2

dydy-jora committed:
ф

я

dydy-jora committed:
я

ф

dydy-jora committed:
123

dydy-jora committed:
Домашняя работа №2

dydy-jora committed:
Домашняя работа №2

Домашняя работа №2

dydy-jora committed:
Поменял файл настройки приложения

dydy-jora committed:
44

kataus commented on lesson-02/pom.xml:
```diff
@@ -1,37 +1,44 @@
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>7</source>
```
7-ка конечно не лучший вариант, но лучше явно указывать используемую версию JMV. Иначе поиск корректной может превратиться в квест для проверяющего

kataus commented on lesson-02/src/main/java/tst/ls_01/service/HomeWork01.java:
```diff
@@ -1,15 +1,15 @@
+    void printResult(String fName, String lName);
+    String readRes(Resource res);
+    void startSurvey(String qv);
+    String getFileData();
```
Не лучший вариант засовывать всё в один bean. Получение вопросов стоит вынести в отдельный dao слой, который уже связать непосредственно с исполнителем теста

kataus commented on lesson-02/src/main/java/tst/ls_01/service/HomeWork01Impl.java:
```diff
@@ -0,0 +1,125 @@
+    public HomeWork01Impl(){ }
+
+    /* получаем имя файла конфигурации */
+    public String getFileData(){
```
Не пропускайте пожалуйста аннотацию @Override

kataus commented on lesson-02/src/main/java/tst/ls_01/service/HomeWork01Impl.java:
```diff
@@ -0,0 +1,125 @@
+    @Autowired
+    private MessageSource ms;
+
+    public HomeWork01Impl(){ }
```
В условии ДЗ бины связываются в конструкторе

dydy-jora committed:
ДЗ №2

Бины связал в конструкторе. Опрос вынес в отдельный бин. Вернул аннотацию @override

dydy-jora committed:
Добавил описание ДЗ

kataus commented on lesson-02/src/main/java/tst/ls_01/Main.java:
```diff
@@ -0,0 +1,26 @@
+
+        hw = (HomeWork01Impl)context.getBean("homeWork01");
+        helper = (DaoHelper) context.getBean("helper");
+        hw.runme(context, helper);
```
Эту связку надо сделать средствами Spring Framework или сделать объединяющий bean который сделает всю работу

dydy-jora committed:
ДЗ №2

kataus commented on lesson-02/src/main/resources/spring-context.xml:
```diff
@@ -1,9 +1,9 @@
-        <constructor-arg name="fileName" value="/data.csv"/>
-    </bean>
-</beans>
+<?xml version="1.0" encoding="UTF-8" ?>
```
Так как мы перешли на аннотации - лучше бы этот файл ушел из нашей жизни
